Updating Gateway Intents Page (#230)
NilPointer-Software/bdfd-wiki: src/guides/gatewayIntents.md

# Gateway Intents
In this guide, you will learn about Discord's Gateway Intents and how to enable them in BDFD.

## What are Gateway Intents?
When a bot is connected to the [Discord Gateway](https://discord.com/developers/docs/topics/gateway), it receives events on actions happening on Discord.
Bots can receive a large amount of events from Discord, so in order to decrease the amount of events each bot receives, Discord requires bots to send Gateway Intents when connecting to the Gateway.\
Gateway Intents allow to specify which events the bot wants to receive.

Gateway Intents can be further divided into:
- Standard Intents[^1]
- [Privileged Intents](#privileged-gateway-intents)

For example, a bot needs to send the `GUILD_MESSAGES` standard intent and the `MESSAGE_CONTENT` privileged intent to receive the `MESSAGE_CREATE` event with the messsage content.

> 📚 Learn more about Gateway Intents [here](https://discord.com/developers/docs/topics/gateway#gateway-intents).

### Privileged Gateway Intents

> Privileged Gateway Intents are special intents which needs to be manually enabled in your bot application settings.

Reviewer: ```suggestion
Privileged Gateway Intents are special intents which need to be manually enabled in your bot application settings.
```

If you are looking for general information about Discord's Gateway Intents, read it [here](https://discord.com/developers/docs/topics/gateway#gateway-intents).

## How to enable Gateway Intents
1. Before starting, make sure that you have installed a version of the app not lower than 2.2.2.
2. After that, follow the instruction below.

![Instruction](https://user-images.githubusercontent.com/70456337/199396053-706bc3a5-fc19-4f03-b40f-9cf13755750c.gif)

## Use of Gateway Intents
### Presence Intent
Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage) and [`$getUserStatus[]`](../bdscript/getUserStatus.md) to work.
### Server Members Intent
Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md))
### Message Content Intent

> Required for the bot to receive information about the message (such as embed data).\

Reviewer 2: You can still receive the basic message information through the `MESSAGE_CREATE` gateway event, but you need this intent to receive the content data stored in the message.
```suggestion
Required for the bot to receive information stored in the message (such as message content or embed data).\
```

It's also required for text commands to work, otherwise your bot can be based only on slash commands. 

> 

Reviewer 3: ```suggestion
```

If you are looking for general information about Discord's Gateway Intents, read it [here](https://discord.com/developers/docs/topics/gateway#gateway-intents).

> ## How to enable Gateway Intents

Reviewer 3: ```suggestion
## Enabling Gateway Intents
```

Reviewer 3: Shortening of Headers as stated in [CONTRIBUTING.md](https://github.com/NilPointer-Software/bdfd-wiki/blob/dev/CONTRIBUTING.md#shortening-headers) file

1. Before starting, make sure that you have installed a version of the app not lower than 2.2.2.
2. After that, follow the instruction below.

![Instruction](https://user-images.githubusercontent.com/70456337/199396053-706bc3a5-fc19-4f03-b40f-9cf13755750c.gif)

> ## Use of Gateway Intents

Reviewer 3: ```suggestion
## Privileged Gateway Intents
```

### Presence Intent

> Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage), [`$getUserStatus[]`](../bdscript/getUserStatus.md), and [`$getCustomStatus[]`](../bdscript/getCustomStatus.md) to work.\

Reviewer 3: ```suggestion
Allows the bot to receive `PRESENCE_UPDATE` event. This intent is primarily used to retrieve current user's presence state. For example, Activities (i.e PLAYING, LISTENING), Presence (i.e Online, Idle) and Custom status.

Functions which requires this intent are as follows:
- [`$getCustomStatus[]`](../bdscript/getCustomStatus.md)
- [`$getUserStatus[]`](../bdscript/getUserStatus.md)
- [`$membersCount[]`](../bdscript/membersCount.md#second-usage)
```

> Allows bot to get information about users in the guild (such as current status).

Reviewer 3: ```suggestion
```

### Server Members Intent

> Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md)).\

Reviewer 3: ```suggestion
Allows the bot to receive `GUILD_MEMBER_ADD`, `GUILD_MEMBER_UPDATE, `GUILD_MEMBER_REMOVE`, and `THREAD_MEMBERS_UPDATE` events. This intent is primarily required to fetch entire list of guild members and getting specific guild member info (like guild joining, leaving, profile update etc.).

Callbacks which requires this intent are as follows:
- [`$onJoined[]`](../callbacks/onJoined.md)
- [`$onLeave[]`](../callbacks/onLeave.md)
```

> Allows bot to get a list of users in the guild.

Reviewer 3: ```suggestion
```
